npms/skale-ima/index.mjs

```diff
@@ -0,0 +1,7638 @@
+    details.write( cc.debug( "TM - sending tx " ) + cc.j( tx ) +
+        cc.debug( " ts: " ) + cc.info( currentTimestamp() ) + "\n" );
+    const id = tmMakeId( details );
+    const score = tmMakeScore( priority );
+    const record = tmMakeRecord( tx, score );
+    details.write(
+        cc.debug( "TM - Sending score: " ) + cc.info( score ) +
+        cc.debug( ", record: " ) + cc.info( record ) +
+        "\n" );
+    const expiration = 24 * 60 * 60; // 1 day;
+    await redis.multi()
+        .set( id, record, "EX", expiration )
+        .zadd( g_tmPool, score, id )
+        .exec();
+    return id;
+}
+
+function tmIsFinished( record ) {
+    if( record == null )
+        return null;
+    return [ "SUCCESS", "FAILED", "DROPPED" ].includes( record.status );
+}
+
+async function tmGetRecord( txId ) {
+    const r = await redis.get( txId );
+    if( r != null )
+        return JSON.parse( r );
+    return null;
+}
+
+async function tmWait( details, txId, ethersProvider, nWaitSeconds = 36000 ) {
+    const strPrefixDetails = cc.debug( "(gathered details)" ) + " ";
+    const strPrefixLog = cc.debug( "(immediate log)" ) + " ";
+    let strMsg =
+        cc.debug( "TM - will wait TX " ) + cc.info( txId ) +
+        cc.debug( " to complete for " ) + cc.info( nWaitSeconds ) +
+        cc.debug( " second(s) maximum" );
+    details.write( strPrefixDetails + strMsg + "\n" );
+    log.write( strPrefixLog + strMsg + "\n" );
+    const startTs = currentTimestamp();
+    while( ! tmIsFinished( await tmGetRecord( txId ) ) &&
+                ( currentTimestamp() - startTs ) < nWaitSeconds )
+        await sleep( 500 );
+    const r = await tmGetRecord( txId );
+    strMsg = cc.debug( "TM - TX " ) + cc.info( txId ) + cc.debug( " record is " ) +
+        cc.info( JSON.stringify( r ) );
+    details.write( strPrefixDetails + strMsg + "\n" );
+    log.write( strPrefixLog + strMsg + "\n" );
+    if( ( !r ) ) {
+        strMsg = cc.error( "TM - TX " ) + cc.info( txId ) + cc.error( " status is " ) +
+            cc.warning( "NULL RECORD" );
+        details.write( strPrefixDetails + strMsg + "\n" );
+        log.write( strPrefixLog + strMsg + "\n" );
+    } else if( r.status == "SUCCESS" ) {
+        strMsg = cc.success( "TM - TX " ) + cc.info( txId ) + cc.success( " success" );
+        details.write( strPrefixDetails + strMsg + "\n" );
+        log.write( strPrefixLog + strMsg + "\n" );
+    } else {
+        strMsg = cc.error( "TM - TX " ) + cc.info( txId ) + cc.error( " status is " ) +
+            cc.warning( r.status );
+        details.write( strPrefixDetails + strMsg + "\n" );
+        log.write( strPrefixLog + strMsg + "\n" );
+    }
+    if( ( !tmIsFinished( r ) ) || r.status == "DROPPED" ) {
+        log.write( cc.error( "TM - TX " ) + cc.info( txId ) +
+            cc.error( " was unsuccessful, wait failed" ) + "\n" );
+        return null;
+    }
+    const joReceipt = await safeGetTransactionReceipt( details, 10, ethersProvider, r.tx_hash );
```

Comment:
Change to CamelCase, please

Reply:
`r.tx_hash` is not our... we do not own it